exporters/prometheus/exporter.go

```diff
@@ -47,13 +47,16 @@ const (
 
 var scopeInfoKeys = [2]string{"otel_scope_name", "otel_scope_version"}
 
-// Exporter is a Prometheus Exporter that embeds the OTel metric.Reader
-// interface for easy instantiation with a MeterProvider.
+// Exporter is a Prometheus Exporter that registers a prometheus.Collector
+// and encapsulates metric.Reader used for integration with a MeterProvider.
 type Exporter struct {
-	metric.Reader
+	reader metric.Reader
 }
 
-var _ metric.Reader = &Exporter{}
+// Reader returns the exporter's metrics reader.
+func (e *Exporter) Reader() metric.Reader {
```

Comment:
Does this still allow the Reader to be accessed and modified? (from https://github.com/open-telemetry/opentelemetry-go/issues/4310)

Reply:
> Does this still allow the Reader to be accessed 

Yes. Via `reader.Reader()`. I was concerned mostly about possible modification after construction which would then lead to race conditions.

> modified

The whole instance cannot be "swapped". Meaning that the `exporter.reader` cannot be set to another instance after the construction.

Reply:
That makes sense, thanks.

Reply:
I tried to add it to the PR description.